File: 25th/jongmlee/MagicalElebator.java
import java.io.*;
import java.util.*;

class Solution {
	public int solution(int storey) {
		int answer = 0;
		int prev = -1;
		
		while (storey > 0) {
			int digit = storey % 10;
			if (digit < 5) {
				answer += digit;
			} else if (digit > 5) {
				answer += 10 - digit;
				storey += digit;
			} else if (digit == 5) {
				answer += digit;
				if (prev >= 6) {
					storey++;
				}
				if (prev == -1) {
					if ((storey / 10) % 10 >= 5) {
						storey += digit;
					}
				}
			}
			storey /= 10;
			prev = digit;
		}
		return answer;
}
}

File: 25th/jeakim/magical_elebator.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(int storey) {
    int answer = 0;

	while (storey > 0){
		if (storey % 10 == 5 && (storey / 10) % 10 >= 5){
			answer += 5;
			storey += 10;
		}
		else if (storey % 10 > 5){
			answer += 10 - storey % 10;
			storey += 10;
		}
		else
			answer += storey % 10;
		storey /= 10;
	}
    return answer;
}

int main()
{
	cout << solution(555) << endl;
	return (0);
}